tests/multi_ptr/multi_ptr_arithmetic_op.h

```diff
@@ -0,0 +1,417 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides code for multi_ptr arithmetic operators
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_TESTS_MULTI_PTR_ARITHMETIC_OP_H
+#define __SYCLCTS_TESTS_MULTI_PTR_ARITHMETIC_OP_H
+
+#include "../common/common.h"
+
+#include "../common/section_name_builder.h"
+#include "multi_ptr_common.h"
+
+namespace multi_ptr_prefetch_member {
+
+constexpr int expected_val = 42;
+
+/**
+ * @brief Provides functor for verification on multi_ptr arithmetic operators
+ * @tparam T Current data type
+ * @tparam AddrSpaceT sycl::access::address_space enumeration's field
+ * @tparam IsDecoratedT sycl::access::decorated enumeration's field
+ */
+template <typename T, typename AddrSpaceT, typename IsDecoratedT>
+class run_multi_ptr_arithmetic_op_test {
+  static constexpr sycl::access::address_space space = AddrSpaceT::value;
+  static constexpr sycl::access::decorated decorated = IsDecoratedT::value;
+  using multi_ptr_t = sycl::multi_ptr<T, space, decorated>;
+
+ public:
+  /**
+   * @param type_name Current data type string representation
+   * @param address_space_name Current sycl::access::address_space string
+   *        representation
+   * @param is_decorated_name Current sycl::access::decorated string
+   *        representation
+   */
+  void operator()(const std::string &type_name,
+                  const std::string &address_space_name,
+                  const std::string &is_decorated_name) {
+    constexpr size_t array_size = 10;
+    // Array that will be used in multi_ptr
+    T arr[array_size];
+    for (size_t i = 0; i < array_size; ++i) {
+      arr[i] = i;
+    }
+    size_t middle_elem_index = array_size / 2;
+    // Expected value that will be returned after multi_ptr's operator will be
+    // called
+    T op_return_result_expected_val;
+    // Expected value that will be stored into multi_ptr after
+    // multi_ptr's operator will be called
+    T op_calling_expected_val;
+
+    auto queue = sycl_cts::util::get_cts_object::queue();
+    SECTION(section_name("Check multi_ptr operator++(multi_ptr& mp)")
```

Comment:
The code of all sections are exactly the same with two exceptions:
1. Expected results
2. called multi_ptr operator

We can avoid code duplication if we create a function with corresponding parameters to specific expected results + a functor calling multi_ptr operator.

Reply:
duplicated code has been moved into a separate function in this commit: https://github.com/KhronosGroup/SYCL-CTS/pull/353/commits/8d9d5dd663cecdf84c90a03c8fa573b70f5a9a6d